eZ/Publish/API/Repository/Tests/SetupFactory/LegacySolr.php

```diff
@@ -104,19 +104,29 @@ protected function indexAll()
         $stmt->execute();
 
         $contentObjects = array();
+        $locations = array();
         while ( $row = $stmt->fetch( PDO::FETCH_ASSOC ) )
         {
             $contentObjects[] = $persistenceHandler->contentHandler()->load(
                 $row['id'],
                 $row['current_version']
             );
+            $locations = array_merge(
+                $locations,
+                $persistenceHandler->locationHandler()->loadLocationsByContent( $row["id"] )
+            );
         }
 
         /** @var \eZ\Publish\Core\Search\Solr\Content\Handler $contentSearchHandler */
         $contentSearchHandler = $searchHandler->contentSearchHandler();
-        $contentSearchHandler->setCommit( false );
-        $contentSearchHandler->purgeIndex();
         $contentSearchHandler->setCommit( true );
```

Comment:
Don't need to set it at all anymore as it is true by default.
Either way what to aim for is that batch purge and insert operations commit should happen at the end, otherwise we are slowing it down quite a bit.